src/pages/ReferralDetailsPage.js

```diff
@@ -0,0 +1,110 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import Button from '@components/Button';
+import CopyTextToClipboard from '@components/CopyTextToClipboard';
+import FixedFooter from '@components/FixedFooter';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Icon from '@components/Icon';
+import {PaymentHands} from '@components/Icon/Illustrations';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import TextLink from '@components/TextLink';
+import useLocalize from '@hooks/useLocalize';
+import Navigation from '@libs/Navigation/Navigation';
+import styles from '@styles/styles';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import NotFoundPage from './ErrorPage/NotFoundPage';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** The type of the content from where CTA was called */
+            contentType: PropTypes.string,
+        }),
+    }).isRequired,
+
+    /** The details about the account that the user is signing in with */
+    account: PropTypes.shape({
+        /** Whether or not a sign on form is loading (being submitted) */
+        loading: PropTypes.bool,
+
+        /** Whether or not the account is validated */
+        validated: PropTypes.bool,
+
+        /** The primaryLogin associated with the account */
+        primaryLogin: PropTypes.string,
+    }),
+};
+
+const defaultProps = {
+    account: null,
+};
+
+function ReferralDetailsPage({route, account}) {
+    const {translate} = useLocalize();
+    const {contentType} = route.params;
+    if (!_.includes(_.values(CONST.REFERRAL_PROGRAM.CONTENT_TYPES), contentType)) {
+        return <NotFoundPage />;
```

Comment:
What if we just defaulted to the "refer a friend" page in this scenario, rather than not found?